### PR TITLE
refactor for lodash usage + deinitialize 

### PR DESCRIPTION
**Pull request by plandem**
*refactor for lodash usage
+added deinitialize() logic to componentWillUnmount

### CODE REVIEW

**Comment by acjay** [comment]
Sorry for the delay -- I'm going to give this a test just to make sure it works with application that depends on it, and if so, I'll go ahead and merge!

**Comment by plandem** [comment]
don't forget about **disaggregateSuperSelector**, library is not using it, so I commented it.

**Comment by acjay** [comment]
I'm gonna try try try to test this with our app in the next day or two

**Comment by plandem** [comment]
still no any news?! :)

**Comment by damassi** [comment]
@plandem - Why the decision to switch to `lodash` and move away from `ramda`? If its to reduce the bundle size via plugin there is https://www.npmjs.com/package/babel-plugin-ramda which is essentially the same thing. IMO I find Ramda to be far more readable due to the data-last quality of it, and also more flexible due to currying.

**Comment by plandem** [comment]
e.g. redux uses lodash, so less differ dependencies for lib.

**Comment by damassi** [comment]
I'll let @acjay chime in here, but I could be open to seeing this refactored using the new `import { select } from 'lodash/fp'` style [now baked into the core lib](https://github.com/lodash/lodash/wiki/FP-Guide), but I'm somewhat hesitant to switch the arguments around so that data comes first because then it looses some of its flexibility.

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "presets": ["es2015-loose", "react"],
-  "plugins": []
+  "presets": ["es2015", "stage-0", "react"],
```
Comment (damassi) [inline .babelrc:2]: Is there a reason for `stage-0` use? That stage is typically reserved for experimental features that might not make it out of committee.

File: src/index.js [hunk 2]
```diff
@@ -129,15 +133,22 @@ export function controller(RootComponent, controllerGenerators, selectorBundles,
...
+        this.controllerMethods.deinitialize();
+      }
     }
 
     getChildContext() {
       // Rather than injecting all of the RootComponent props into the context,
-      // we only explictly pass selector and controller method props.
-      const selectorProps = R.pick(R.keys(selectorPropTypes), this.props);
-      const childContext = R.merge(selectorProps, this.controllerMethods);
-      return childContext;
```
Comment (damassi) [inline src/index.js:140]: Keeping an explicit return variable makes things easier to debug and also makes the code more readable ("the child context consists of these selector props and controller methods")

File: src/selector_utils.js [hunk 3]
```diff
@@ -57,9 +57,11 @@ export function aggregateSelectors(bundle) {
  *   annotated with a propType property.
  */
 export function disaggregateSuperSelector(superSelector) {
-  return R.mapObjIndexed((propType, propName) => {
+//TODO: refactor for lodash
+/*  return R.mapObjIndexed((propType, propName) => {
```
Comment (damassi) [inline src/selector_utils.js:61]: Why comment out with a `TODO`?

**Pull request closed without merging**